src/ansys/simai/core/data/model_configuration.py

```diff
@@ -0,0 +1,257 @@
+
+    Example:
+        Define a new configuration and launch a build.
+
+        .. code-block:: python
+
+            import ansys.simai.core as asc
+            from ansys.simai.core.data.model_configuration import (
+                ModelConfiguration,
+                ModelInput,
+                ModelOutput,
+            )
+
+            simai = asc.from_config()
+
+            # Get the project of interest
+            aero_dyn_project = simai.projects.get(name="aero-dyn")
+
+            # Define the inputs of the model
+            model_input = ModelInput(surface=["Velocity"], boundary_conditions=["Vx"])
+
+            # Define the outputs of the model
+            model_output = ModelOutput(
+                surface=["Pressure", "WallShearStress_0"], volume=["Velocity_0", "Pressure"]
+            )
+
+            # Define the model coefficient
+            global_coefficients = [("max(Pressure)", "maxpress")]
+
+            # Define the build configuration for the model
+            new_conf = ModelConfiguration(
+                project=aero_dyn_project,
+                build_preset="debug",
+                continuous=False,
+                input=model_input,
+                output=model_output,
+                global_coefficients=global_coefficients,
+                simulation_volume=simulation_volume,
+            )
+
+            # Launch a mode build with the new configuration
+            new_model = simai.models.build(new_conf)
+    """
+
+    project: "Project" = None
+    build_preset: Literal[
+        "debug",
+        "short",
+        "default",
+        "long",
+    ] = "default"
+    continuous: bool = False
+    input: ModelInput = ModelInput()
+    output: ModelOutput = ModelOutput()
+
+    def __set_gc(self, gcs: list[tuple | GlobalCoefficientUnit]):
+        verified_gcs = []
+        for gc in gcs:
+            gc_unit = GlobalCoefficientUnit(*gc) if isinstance(gc, tuple) else gc
+            self.project.verify_gc_formula(
+                gc_unit.formula, self.input.boundary_conditions, self.output.surface
+            )
+            verified_gcs += [gc_unit]
+        self.__dict__["global_coefficients"] = verified_gcs
+
+    def __get_gc(self):
+        return self.__dict__.get("global_coefficients")
+
+    global_coefficients: ModelOutput = property(__get_gc, __set_gc)
+
+    def __init__(
+        self,
+        boundary_conditions: Optional[dict[str, Any]] = None,
+        build_preset: Optional[str] = None,
+        continuous: Optional[bool] = False,
+        fields: Optional[dict[str, Any]] = None,
+        global_coefficients: Optional[dict[dict[str, Any]] | GlobalCoefficientUnit] = None,
+        simulation_volume: Optional[dict[str, Any]] = None,
+        project: Optional["Project"] = None,
+        input: Optional[ModelInput] = None,
+        output: Optional[ModelOutput] = None,
+    ):
+        """Sets the properties of a build configuration."""
+        self.project = project
```

Comment:
so self.project is None and stuff like `self.project.compute_gc_formula` crashes

Reply:
Correct, which it is fair considering that a build configuration is tight to a project. Regardless, I set some checks and raise exception in `compute_gc_formula` for highlighting it to the user.

Reply:
Ok but why is project optional then ? (`project: Optional["Project"] = None`)

Reply:
to give more flexibility to the user to gradually compose a configuration, e.g., first create inputs, then outputs, etc. To check the gcs, however, a project should be defined.